src/test/kotlin/cacophony/automata/DeterminizationTest.kt

```diff
@@ -0,0 +1,187 @@
+package cacophony.automata
+
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+class DeterminizationTest {
+    private fun makeNFA(
+        starting: Int,
+        productions: Map<Pair<Int, Char>, List<Int>>,
+        epsilonProductions: Map<Int, List<Int>>,
+        accepting: Int,
+    ): NFA<Int> =
+        object : NFA<Int> {
+            private val allStates =
+                (
+                    setOf(starting, accepting) union
+                        productions.keys
+                            .unzip()
+                            .first
+                            .toSet() union productions.values.flatten().toSet() union epsilonProductions.keys union
+                        epsilonProductions.values
+                            .flatten()
+                            .toSet()
+                ).toList()
+
+            override fun getStartingState() = starting
+
+            override fun getProductions() = productions
+
+            override fun getProductions(
+                state: Int,
+                symbol: Char,
+            ) = productions[state to symbol] ?: emptyList()
+
+            override fun getEpsilonProductions() = epsilonProductions
+
+            override fun getAcceptingState() = accepting
+
+            override fun isAccepting(state: Int) = state == accepting
+
+            override fun getAllStates() = allStates
+        }
+
+    @Test
+    fun `Empty language NFA maps to empty language DFA`() {
```

Comment:
I think we started with naming test functions with lowercase first letter, so maybe it's the way to go, to keep one style